tests/30rooms/60version_upgrade.pl

```diff
@@ -384,55 +381,50 @@ sub upgrade_room_synced {
             },
          );
       })->then( sub {
-         matrix_do_and_wait_for_sync(
-            $creator,
-            do => sub {
-               upgrade_room(
-                  $creator, $room_id, new_version => $TEST_NEW_VERSION,
-               );
-            },
-            check => sub {
-               my ( $sync_body, $new_room_id ) = @_;
-
-               log_if_fail "sync body", $sync_body;
-
-               my $joined = $sync_body->{rooms}{join} // {};
-               foreach my $room_id ( keys %{ $joined } ) {
-                  foreach my $ev ( @{ $joined->{$room_id}{timeline}{events} } ) {
-                     $received_events{$room_id}->{$ev->{type}}->{$ev->{state_key} // ''} = $ev;
-                  }
-               }
-
-               # wait until we've received both alias events in both rooms.
-               foreach my $room_id ( $room_id, $new_room_id ) {
-                  my $rs = $received_events{$room_id} // {};
-                  foreach ( qw( m.room.aliases m.room.canonical_alias ) ) {
-                     log_if_fail "$_ in $room_id", $rs->{$_};
-                     return 0 unless $rs->{$_};
-                  }
-               }
-               return 1;
+         upgrade_room_synced(
+            $creator, $room_id,
+            new_version => $TEST_NEW_VERSION,
+            expected_event_counts => {
+               'm.room.aliases' => 1, 'm.room.canonical_alias' => 1,
             },
          );
       })->then( sub {
-         my ( $new_room_id, $sync_body ) = @_;
+         ( $new_room_id ) = @_;
 
-         my $old_aliases = $received_events{$room_id}->{'m.room.aliases'}->{$server_name};
-         assert_deeply_eq( $old_aliases->{content}, {aliases => []}, "aliases on old room" );
+         matrix_get_room_state(
+            $creator, $room_id,
+            type=>'m.room.aliases', state_key=>$server_name,
+         );
```

Comment:
Any reason for using `matrix_get_room_state` rather than just syncing?

Reply:
I decided it was going to be a pain to find the right events in the sync. Could change it though.

Reply:
Nah, its fine